### PR TITLE
postgresql (PostgreSQL): rework for accidental 16.x upgrade

### PR DESCRIPTION
**Pull request by MingcongBai**
Topic Description
-----------------

- p-vector: rebuild for accidental PostgreSQL upgrade
- groups/postgresql-rebuilds: add p-vector
    Not a library dependency but required a PostgreSQL database template to build.
- libreoffice: rebuild for accidental PostgreSQL upgrade
- qt-6: rebuild for accidental PostgreSQL upgrade
- qt-4: rebuild for accidental PostgreSQL upgrade
- qt-5: rebuild for accidental PostgreSQL upgrade
- intel-oneapi-basekit: rebuild for accidental PostgreSQL upgrade
- redland: rebuild for accidental PostgreSQL upgrade
- sysbench: rebuild for accidental PostgreSQL upgrade
- soci: rebuild for accidental PostgreSQL upgrade

... and 16 more commits

Package(s) Affected
-------------------

- amarok-trinity: 14.1.2-1
- apr-util: 1.6.3-1
- dovecot: 2.3.10.1-6
- gdal: 3.8.4-3
- intel-oneapi-basekit: 2024.2.0-1
- libdbi-drivers: 0.9.0-1
- libgda: 6.0.0-1
- libpqxx: 6.4.5-2
- libreoffice: 24.2.3.2-2
- minetest: 5.7.0+irrlicht1.9.0mt10-1
- p-vector: 0.4.1-1
- php: 1:8.3.8-2
- php7: 1:0-2
- postfix: 3.7.3-6
- postgresql: 1:13.15
- psycopg2: 2.8.6-2
- qt-4: 4.8.7-21
- qt-5: 1:5.15.13+webengine5.15.16+webkit5.212.0+kde20240408-6
- qt-5-doc: 1:5.15.13+webengine5.15.16+webkit5.212.0+kde20240408-6
- qt-6: 6.7.2-2
- redland: 1.0.17-9
- ruby-pg: 1.1.4-4
- soci: 4.0.1-1
- sysbench: 1.0.20-1
- tqt3: 14.1.2-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit postgresql groups/postgresql-rebuilds
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: app-database/postgresql/autobuild/defines [hunk 1]
```diff
@@ -34,3 +34,6 @@ ABMK="world"
 ABSHADOW=0
 RECONF=0
 NOSTATIC=0
+
+# FIXME: PostgreSQL accidentally shipped a 16.x version. Bumped REL to reset.
```
Comment (KexyBiscuit) [inline app-database/postgresql/autobuild/defines]:
```suggestion
# FIXME: PostgreSQL accidentally shipped a 16.x version. Bumped EPOCH to reset.
```

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]

File: app-database/postgresql/autobuild/defines (hunk 1 above)
Comment (MingcongBai) [reply app-database/postgresql/autobuild/defines]: I will wait for the current build to pass.

**Review by Cyanoxygen — APPROVED** [review]

**Comment by Cyanoxygen** [comment]
I can not do the multi-version packaging right now, approving.

**Comment by MingcongBai** [comment]
> I can not do the multi-version packaging right now, approving.

Pending build tests.

**Comment by Cyanoxygen** [comment]
Testing
--------

Tested on amd64.

Here's my PostgreSQL server running:

image

And a Forgejo instance which is connected to that PostgreSQL:

image

**Review by KexyBiscuit — APPROVED** [review]

**Comment by MingcongBai** [comment]
RISC-V build completed manually after network failure.

**Pull request merged**